aptos-move/framework/aptos-stdlib/sources/cryptography/algebra.move

```diff
@@ -16,14 +48,154 @@ module aptos_std::algebra {
     // Public functions begin.
     //
 
-    /// Compute `x + y` for elements `x` and `y` of an algebraic structure `S`.
+    /// Check if `x == y` for elements `x` and `y` of a structure `S`.
+    public fun eq<S>(x: &Element<S>, y: &Element<S>): bool {
+        abort_unless_cryptography_algebra_natives_enabled();
+        eq_internal<S>(x.handle, y.handle)
+    }
+
+    /// Convert a u64 to an element of a structure `S`.
+    public fun from_u64<S>(value: u64): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: from_u64_internal<S>(value)
+        }
+    }
+
+    /// Return the additive identity of field `S`, or the identity of group `S`.
+    public fun zero<S>(): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: zero_internal<S>()
+        }
+    }
+
+    /// Return the multiplicative identity of field `S`, or a fixed generator of group `S`.
+    public fun one<S>(): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: one_internal<S>()
+        }
+    }
+
+    /// Compute `-x` for an element `x` of a structure `S`.
+    public fun neg<S>(x: &Element<S>): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: neg_internal<S>(x.handle)
+        }
+    }
+
+    /// Compute `x + y` for elements `x` and `y` of structure `S`.
     public fun add<S>(x: &Element<S>, y: &Element<S>): Element<S> {
         abort_unless_cryptography_algebra_natives_enabled();
         Element<S> {
             handle: add_internal<S>(x.handle, y.handle)
         }
     }
 
+    /// Compute `x - y` for elements `x` and `y` of a structure `S`.
+    public fun sub<S>(x: &Element<S>, y: &Element<S>): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: sub_internal<S>(x.handle, y.handle)
+        }
+    }
+
+    /// Compute `x * y` for elements `x` and `y` of a structure `S`.
+    public fun mul<S>(x: &Element<S>, y: &Element<S>): Element<S> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<S> {
+            handle: mul_internal<S>(x.handle, y.handle)
+        }
+    }
+
+    /// Try computing `x / y` for elements `x` and `y` of a structure `S`.
+    /// Return none if y equals to `zero<S>()`.
```

Comment:
Here, you are assuming `S` to be a field where only the additive identity is non-invertible. So anything else but the identity can be used as a denominator. But in other structures like rings, some elements might be invertible while others might not (e.g., see [integer rings](https://en.wikipedia.org/wiki/Unit_(ring_theory)#Integer_ring)).

Instead, the correct thing to say here would be:

```
Return None if `y` does not have a multiplicative inverse in the structure `S`.
```

Reply:
Fixed.